### PR TITLE
Random cleaning of SpiConnectionSettings

### PR DESCRIPTION
**Pull request by shaggygi**
Removed private fields & moved to default property values, added spaces, and a misspelling in a pear tree.

### CODE REVIEW

File: src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs [hunk 1]
```diff
@@ -11,10 +11,6 @@ namespace System.Device.Spi
     /// </summary>
     public sealed class SpiConnectionSettings
```
Comment (krwq) [inline src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs]: I wonder if we should rename this to `HardwareSpiSettings`, some of these options don't apply for SoftwareSpi

Comment (shaggygi) [reply src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs]: They are kind of a mixture.  Better yet... how about just SpiSettings (also I2cSettings)?

File: src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs [hunk 2]
```diff
@@ -26,35 +22,38 @@ public SpiConnectionSettings(int busId, int chipSelectLine)
...
+        /// <summary>
+        /// The SPI mode being used.
+        /// </summary>
+        public SpiMode Mode { get; set; } = SpiMode.Mode0;
+
         /// <summary>
         /// The length of the data to be transfered.
         /// </summary>
-        public int DataBitLength { get; set; }
+        public int DataBitLength { get; set; } = 8;
```
Comment (joperezr) [inline src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs]: NIT: can we also have the comment moved mentioning that this represents 1 byte?

File: src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs (hunk 1 above)
Comment (joperezr) [reply src/System.Device.Gpio/System/Device/Spi/SpiConnectionSettings.cs]: I don't think we should rename this. At least not to HardwareSpiSettings, SpiSettings might be an option, but honestly I don't think that the current names are wrong, plus a good thing is that they are the same in WinRT, so having parity may be better.

**Comment by joperezr** [comment]
Thanks @shaggygi

**Pull request merged**